src/App.js

```diff
@@ -1,41 +1,72 @@
 import "./App.css";
 import logo from "./logo.png";
+import { app } from "firebase.config";
+import { Route, Routes } from "react-router";
+import HomePage from "pages/HomePage/HomePage";
+import Main from "Main";
+import ExplorePage from "pages/ExplorePage/ExplorePage";
+import ProfilePage from "pages/ProfilePage/ProfilePage";
+import LoginPage from "pages/LoginPage/LoginPage";
+import SignUpPage from "pages/SignUpPage/SignUpPage";
+import BookmarkPage from "pages/BookmarkPage/BookmarkPage";
+import RequiredAuth from "components/common/PrivatedRoutes/RequiredRoutes";
+import Mockman from 'mockman-js';
+import NotFoundPage from "pages/NotFoundPage/NotFoundPage";
+import { ROUTES } from "utils/routes";
+import { Suspense } from "react";
 
+console.log(app);
 function App() {
   return (
     <div className="App">
-      <header className="App-header">
-        <img src={logo} alt="mockBee logo" width="180" height="180" />
-        <h1 className="brand-title">
-          Welcome to <span>mockBee!</span>
-        </h1>
-        <p className="brand-description">
-          Get started by editing <code>src/App.js</code>
-        </p>
-        <div className="links">
-          <a
-            href="https://mockbee.netlify.app/"
-            target="_blank"
-            rel="noreferrer"
-          >
-            Explore mockBee
-          </a>
-          <a
-            href="https://mockbee.netlify.app/docs/api/introduction"
-            target="_blank"
-            rel="noreferrer"
-          >
-            API Documentation
-          </a>
-          <a
-            href="https://github.com/neogcamp/mockBee"
-            target="_blank"
-            rel="noreferrer"
-          >
-            Contribute
-          </a>
-        </div>
-      </header>
+       <Routes >
+        <Route element={<Main />}>
+          <Route path={ROUTES.ROUTE_PATH_HOMEPAGE} element={
+            <Suspense fallback={"..loading"}>
+              <HomePage />
+            </Suspense>}
+          />
+          <Route path={ROUTES.ROUTE_PATH_EXPLOREPAGE} element={
+            <RequiredAuth>
+              <Suspense fallback={"..loading"}>
+                <ExplorePage />
+              </Suspense>
+            </RequiredAuth>}
+          />
+          <Route path={ROUTES.ROUTE_PATH_BOOKMARKPAGE} element={
+            <RequiredAuth>
+              <Suspense fallback={"..loading"}>
+                <BookmarkPage />
+              </Suspense>
+            </RequiredAuth>}
```

Comment:
you can wrap all route which requires auth in single route where you will pass requiresauth component, this way you won't have to repeatedly wrap your private route components in require auth

Reply:
like creating a object and looping but then i have to create 2 different obj of private and public and i guess it will come up with same efforts and not much benefit